Realm/RLMSyncConfiguration.h

```diff
@@ -18,13 +18,40 @@
 
 #import <Foundation/Foundation.h>
 
+@class RLMRealm;
 @class RLMRealmConfiguration;
 @class RLMUser;
 @class RLMApp;
 @protocol RLMBSON;
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**  Determines file behavior during a client reset.
+
+ - see: https://docs.mongodb.com/realm/sync/error-handling/client-resets/
+*/
+typedef NS_ENUM(NSUInteger, RLMClientResetMode) {
+    /// The SDK will create a back up of unsynced data.The client reset error handler may be manually overwritten to
+    /// transfer data from the backup copy to a new destination. Otherwise no effort to transfer the data from the backup is carried out.
+    RLMClientResetModeManual,
+    /// The SDK will overwrite the client database with the server database. Object accessors remain bound so Realm notifications are not disrupted.
+    RLMClientResetModeDiscardLocal
+};
+
+/**
+ A block type used to report before a client reset will occur.
+ The RlMRealm argument contains the local database state prior to client reset.
+ */
+typedef void(^RLMClientResetBeforeBlock)(RLMRealm * _Nonnull);
+
+// ???: Is there really no way to label these arguments?
```

Comment:
Is there really no way to label these arguments? It'd be nice if there were auto filled template in xcode.